instrumentation/kafka-connect-metrics-1.0.0/src/main/java/com/nr/instrumentation/kafka/connect/KafkaConnectMetricsReporter.java

```diff
@@ -0,0 +1,176 @@
+/*
+ *
+ *  * Copyright 2023 New Relic Corporation. All rights reserved.
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package com.nr.instrumentation.kafka.connect;
+
+import com.newrelic.api.agent.NewRelic;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.metrics.KafkaMetric;
+import org.apache.kafka.connect.runtime.ConnectMetrics;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Level;
+
+public class KafkaConnectMetricsReporter implements org.apache.kafka.common.metrics.MetricsReporter {
+
+    private static final String KAFKA_CONNECT_METRICS_EVENT_NAME = "KafkaConnectMetrics";
+
+    private static final String METRIC_PREFIX = "Kafka/Connect/";
+
+    private static final boolean KAFKA_METRICS_DEBUG = NewRelic.getAgent().getConfig().getValue("kafka.metrics.debug.enabled", false);
+
+    private static final boolean METRICS_AS_EVENTS = NewRelic.getAgent().getConfig().getValue("kafka.metrics.as_events.enabled", false);
+
+    private static final long REPORTING_INTERVAL_IN_SECONDS = NewRelic.getAgent().getConfig().getValue("kafka.metrics.interval", 30);
+
+    private static final ScheduledThreadPoolExecutor EXECUTOR = new ScheduledThreadPoolExecutor(1, buildThreadFactory());
+
+    private static final AtomicBoolean INITIALIZED = new AtomicBoolean(false);
+
+    // key is the metric/event name
+    private final Map<String, KafkaMetric> metrics = new ConcurrentHashMap<>();
+
+    private ScheduledFuture<?> scheduledFuture;
+
+    // the metrics are shared between Workers. So only one reporter needs to be attached.
+    public static void initialize(ConnectMetrics.MetricGroup metricGroup) {
```

Comment:
Nice. I'd like to refactor some of the other Kafka metrics reporters in streams/clients to use this approach.